src/base/install/addKeyToKnownHosts.m

```diff
@@ -0,0 +1,61 @@
+function keyAdded = addKeyToKnownHosts(siteName)
+% Checks if the public key to `site.ext` exists
+% If the public key of the `site.ext` does not exist,
+% adds the public key to the known hosts
+%
+% USAGE:
+%
+%   keyAdded = addKeyToKnownHosts()
+%
+% INPUT:
+%   siteName:   Name of the site for which the public key shall be added
+%               (default: `github.com`)
+%
+% OUTPUT:
+%   keyAdded:   Boolean (true if key has been added successfully or exists)
+%
+% .. Author:
+%      - Laurent Heirendt
+
+    % set the default site name
+    if ~exist('siteName', 'var')
+        siteName = 'github.com';
+    end
+
+    % find the keyscan
+    [status_keyscan, result_keyscan] = system('ssh-keyscan');
+
+    % user directory
+    if ispc
+        homeDir = getenv('userprofile');
+    else
+        homeDir = getenv('HOME');
+    end
+
+    if status_keyscan == 1 && ~isempty(strfind(result_keyscan, 'usage:'))
+
+        % touch the file first
+        system(['touch ', homeDir, filesep, '.ssh', filesep, 'known_hosts']);
```

Comment:
Is this the same on windows and unix os?
Also, it is not necessarily present on a windows system (thus you would have to create the folder first).
Not sure what the normal "search path" for known ssh key on a windows system is...

Reply:
It is the same on windows and unix. I tested on both.

As it my not be present, the file is `touched` before. The path is the same on all platforms when using `gitBash`.

Reply:
The path is not present on the Win7 I'm using for testing, which also has git-bash installed. is it possible, that this is again only present if ssh was used for git checkout?

Reply:
It is for sure present when having cloned with `git@github.com`. When the directory `.ssh` is not present and the file `known_hosts` does not exist, it is created when running:
```
touch ~/.ssh/known_hosts
```

Reply:
On Win7, we may need to create the folder first.

Reply:
The reason for my comment seems to stem from an odd system configuration, where the .ssh folder is not in `%userprofile%` but a different folder. 
Not sure, but maybe it would be necessary to check where the .ssh folder actually is to properly set it up.